### ISSUE TITLE
Smoke tests error: Cannot initialize @agoric/assert

### ISSUE THREAD

**Issue by Jorge-Lopes** [alexanderem49/simple-exchange#8]
When the offer build script imports `import { AmountMath } from '@agoric/ertp';` it will trigger the following error:
` Cannot initialize @agoric/assert, missing globalThis.assert methods Fail`

Complete error message:

```
jorgelopes@Jorges-MBP smokeTests % ./buyOffer.sh
(Error#1)
Error#1: Cannot initialize @agoric/assert, missing globalThis.assert methods Fail

  at packages/assert/src/assert.js:48:9
  at async Promise.all (index 0)

Executing  [
  '--node=http://0.0.0.0:26657',
  '--chain-id=agoriclocal',
  '--keyring-backend=test',
  '--from=agoric1dvk4vr2ss65zcls9qdf0ntcaaaunw3cyn6l8e8',
  'tx',
  'swingset',
  'wallet-action',
  '--allow-spend',
  '',
  '--yes'
]
Error: Spend action cannot be empty: unknown request
Usage:
  agd tx swingset wallet-action [json string] [flags]

Flags:
  -a, --account-number uint      The account number of the signing account (offline mode only)
      --allow-spend              Allow the WalletAction to spend assets
  -b, --broadcast-mode string    Transaction broadcasting mode (sync|async|block) (default "sync")
      --dry-run                  ignore the --gas flag and perform a simulation of a transaction, but don't broadcast it (when enabled, the local Keybase is not accessible)
      --fee-account string       Fee account pays fees for the transaction instead of deducting from the signer
      --fees string              Fees to pay along with transaction; eg: 10uatom
      --from string              Name or address of private key with which to sign
      --gas string               gas limit to set per-transaction; set to "auto" to calculate sufficient gas automatically (default 200000)
      --gas-adjustment float     adjustment factor to be multiplied against the estimate returned by the tx simulation; if the gas limit is set manually this flag is ignored  (default 1)
      --gas-prices string        Gas prices in decimal format to determine the transaction fee (e.g. 0.1uatom)
      --generate-only            Build an unsigned transaction and write it to STDOUT (when enabled, the local Keybase is not accessible)
  -h, --help                     help for wallet-action
      --keyring-backend string   Select keyring's backend (os|file|kwallet|pass|test|memory) (default "os")
      --keyring-dir string       The client Keyring directory; if omitted, the default 'home' directory will be used
      --ledger                   Use a connected Ledger device
      --node string              <host>:<port> to tendermint rpc interface for this chain (default "tcp://localhost:26657")
      --note string              Note to add a description to the transaction (previously --memo)
      --offline                  Offline mode (does not allow any online functionality
  -o, --output string            Output format (text|json) (default "json")
  -s, --sequence uint            The sequence number of the signing account (offline mode only)
      --sign-mode string         Choose sign mode (direct|amino-json), this is an advanced feature
      --timeout-height uint      Set a block timeout height to prevent the tx from being committed past a certain height
  -y, --yes                      Skip tx broadcasting prompt confirmation

Global Flags:
      --chain-id string     The network chain ID
      --home string         directory for config and data (default "/Users/jorgelopes/.agoric")
      --log_format string   The logging format (json|plain) (default "plain")
      --log_level string    The logging level (trace|debug|info|warn|error|fatal|panic) (default "info")
      --trace               print out full stack trace on errors

agoric: (Error#1)
Error#1: Command failed: agd --node=http://0.0.0.0:26657 --chain-id=agoriclocal --keyring-backend=test --from=agoric1dvk4vr2ss65zcls9qdf0ntcaaaunw3cyn6l8e8 tx swingset wallet-action --allow-spend  --yes

  at checkExecSyncError (node:child_process:861:11)
  at execFileSync (node:child_process:896:15)
  at execSwingsetTransaction (packages/agoric-cli/src/lib/chain.js:78:12)
  at Command.<anonymous> (packages/agoric-cli/src/commands/wallet.js:125:7)
  at main (packages/agoric-cli/src/main.js:400:19)
```

**Comment by Jorge-Lopes** [comment]
If I make an empty offer, that will no longer require an amount, the error triggered is:

Probably related to https://github.com/Agoric/agoric-sdk/issues/8321

Note: this is not related to the above error
```
2023-10-23T14:07:55.945Z SwingSet: xsnap: v10: UnhandledPromiseRejectionWarning: (Error#3)
2023-10-23T14:07:55.946Z SwingSet: xsnap: v10: Error#3: No inbound handler for wallet
2023-10-23T14:07:55.946Z SwingSet: xsnap: v10: Error: No inbound handler for (a string)
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:243)
 at fail (/bundled-source/.../node_modules/ses/src/error/assert.js:357)
 at baseAssert (/bundled-source/.../node_modules/ses/src/error/assert.js:377)
 at fromBridge (.../vats/src/bridge.js:62)
 at apply ()
 at apply ()
 at In "fromBridge" method of (BridgeScopedManager) (/bundled-source/.../node_modules/@endo/exo/src/exo-tools.js:42)
 at inbound (.../vats/src/bridge.js:151)
 at apply ()
 at apply ()
 at In "inbound" method of (BridgeManagerKit privateInbounder) (/bundled-source/.../node_modules/@endo/exo/src/exo-tools.js:42)
 at apply ()
 at apply ()
 at dispatchToHandler (/bundled-source/.../node_modules/@endo/eventual-send/src/handled-promise.js:163)
 at win (/bundled-source/.../node_modules/@endo/eventual-send/src/handled-promise.js:511)
 at ()
```

**Comment by Jorge-Lopes** [comment]
> When the offer build script imports `import { AmountMath } from '@agoric/ertp';` it will trigger the following error: ` Cannot initialize @agoric/assert, missing globalThis.assert methods Fail`
> 
> Complete error message:
> 
> ```
> jorgelopes@Jorges-MBP smokeTests % ./buyOffer.sh
> (Error#1)
> Error#1: Cannot initialize @agoric/assert, missing globalThis.assert methods Fail
> 
>   at packages/assert/src/assert.js:48:9
>   at async Promise.all (index 0)
> 
> Executing  [
>   '--node=http://0.0.0.0:26657',
>   '--chain-id=agoriclocal',
>   '--keyring-backend=test',
>   '--from=agoric1dvk4vr2ss65zcls9qdf0ntcaaaunw3cyn6l8e8',
>   'tx',
>   'swingset',
>   'wallet-action',
>   '--allow-spend',
>   '',
>   '--yes'
> ]
> Error: Spend action cannot be empty: unknown request
> Usage:
>   agd tx swingset wallet-action [json string] [flags]
> 
> Flags:
>   -a, --account-number uint      The account number of the signing account (offline mode only)
>       --allow-spend              Allow the WalletAction to spend assets
>   -b, --broadcast-mode string    Transaction broadcasting mode (sync|async|block) (default "sync")
>       --dry-run                  ignore the --gas flag and perform a simulation of a transaction, but don't broadcast it (when enabled, the local Keybase is not accessible)
>       --fee-account string       Fee account pays fees for the transaction instead of deducting from the signer
>       --fees string              Fees to pay along with transaction; eg: 10uatom
>       --from string              Name or address of private key with which to sign
>       --gas string               gas limit to set per-transaction; set to "auto" to calculate sufficient gas automatically (default 200000)
>       --gas-adjustment float     adjustment factor to be multiplied against the estimate returned by the tx simulation; if the gas limit is set manually this flag is ignored  (default 1)
>       --gas-prices string        Gas prices in decimal format to determine the transaction fee (e.g. 0.1uatom)
>       --generate-only            Build an unsigned transaction and write it to STDOUT (when enabled, the local Keybase is not accessible)
>   -h, --help                     help for wallet-action
>       --keyring-backend string   Select keyring's backend (os|file|kwallet|pass|test|memory) (default "os")
>       --keyring-dir string       The client Keyring directory; if omitted, the default 'home' directory will be used
>       --ledger                   Use a connected Ledger device
>       --node string              <host>:<port> to tendermint rpc interface for this chain (default "tcp://localhost:26657")
>       --note string              Note to add a description to the transaction (previously --memo)
>       --offline                  Offline mode (does not allow any online functionality
>   -o, --output string            Output format (text|json) (default "json")
>   -s, --sequence uint            The sequence number of the signing account (offline mode only)
>       --sign-mode string         Choose sign mode (direct|amino-json), this is an advanced feature
>       --timeout-height uint      Set a block timeout height to prevent the tx from being committed past a certain height
>   -y, --yes                      Skip tx broadcasting prompt confirmation
> 
> Global Flags:
>       --chain-id string     The network chain ID
>       --home string         directory for config and data (default "/Users/jorgelopes/.agoric")
>       --log_format string   The logging format (json|plain) (default "plain")
>       --log_level string    The logging level (trace|debug|info|warn|error|fatal|panic) (default "info")
>       --trace               print out full stack trace on errors
> 
> agoric: (Error#1)
> Error#1: Command failed: agd --node=http://0.0.0.0:26657 --chain-id=agoriclocal --keyring-backend=test --from=agoric1dvk4vr2ss65zcls9qdf0ntcaaaunw3cyn6l8e8 tx swingset wallet-action --allow-spend  --yes
> 
>   at checkExecSyncError (node:child_process:861:11)
>   at execFileSync (node:child_process:896:15)
>   at execSwingsetTransaction (packages/agoric-cli/src/lib/chain.js:78:12)
>   at Command.<anonymous> (packages/agoric-cli/src/commands/wallet.js:125:7)
>   at main (packages/agoric-cli/src/main.js:400:19)
> ```

The solution was to update the `@endo/init` version on the the contract package.json dependencies
from `^0.5.37`  to `^0.5.56`

**Comment by Jorge-Lopes** [comment]
Fixed [d9bdaab](https://github.com/alexanderem49/simple-exchange/commit/d9bdaab97e622da4e28a8e1a7bb8b69fe35fcf7b)
